src/components/basic/Button.js

```diff
@@ -1,27 +1,43 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import classnames from 'classnames';
+import PropTypes from "prop-types";
+import React from "react";
+import classnames from "classnames";
 
-const Button = (props) => {
-	const { icon, className, type = 'button', text } = props;
-	const buttonCssClasses = classnames('dg_button', className);
-	const iconCssClasses = classnames('dg_button_icon', icon);
-	const buttonIcon = icon ? <i aria-hidden="true" className={iconCssClasses} /> : null;
-	return (
-		<button type={type} className={buttonCssClasses}>
-			{buttonIcon}
-			{text}
-		</button>
-	);
+const Button = props => {
+  const {
+    as: As = "button",
+    icon,
+    className,
+    type = "button",
+    text,
+    ...rest
+  } = props;
+  const buttonCssClasses = classnames("dg_button", className);
+  const iconCssClasses = classnames("dg_button_icon", icon);
+  const buttonIcon = icon ? (
+    <i aria-hidden="true" className={iconCssClasses} />
+  ) : null;
+  return As === "button" ? (
+    <As type={type} className={buttonCssClasses}>
+      {buttonIcon}
+      {text}
+    </As>
+  ) : (
+    <As className={buttonCssClasses} {...rest}>
+      {buttonIcon}
+      {text}
+    </As>
+  );
 };
 
 Button.propTypes = {
-	/** Button Text */
-	text: PropTypes.string.isRequired,
-	/** Font Awesome icon class */
-	icon: PropTypes.string,
-	/** Button Type - either `button`, `submit`, `reset` */
-	type: PropTypes.string
+  /** Button Text */
+  text: PropTypes.string.isRequired,
+  /** Font Awesome icon class */
+  icon: PropTypes.string,
+  /** Button Type - either `button`, `submit`, `reset` */
+  type: PropTypes.string,
```

Comment:
Just thinking out loud, should this be set as required? I'm assuming this defaults to button if there is no `type` property assigned to it.

Reply:
since it can be a link which does not have a type, it can't be required (at least i dont think).

Reply:
this quesiton, probably is a good, should this be it's own component or not question.

Reply:
ended up removing text decoration from any state for a secondary button.